pkg/util/prefetch/reader.go

```diff
@@ -88,7 +100,12 @@ func (r *Reader) Read(data []byte) (int, error) {
 	}
 }
 
-// Close implements io.Closer.
+// Close implements io.Closer. Close should not be called concurrently with Read.
 func (r *Reader) Close() error {
-	return r.r.Close()
+	ret := r.r.Close()
```

Comment:
not sure if the undering reader is safe to call `close/read` concurrently

Reply:
I think we can't assume it so I added the comment